deepspeed/pt/deepspeed_dataloader.py

```diff
@@ -16,13 +16,15 @@ def __init__(self,
                  tput_timer,
                  collate_fn=None,
                  num_local_io_workers=None,
-                 data_sampler=None):
+                 data_sampler=None,
+                 data_parallel_world_size=None,
+                 data_parallel_rank=None):
         self.tput_timer = tput_timer
         self.batch_size = batch_size
 
         if local_rank >= 0:
             if data_sampler is None:
-                data_sampler = DistributedSampler(dataset)
+                data_sampler = DistributedSampler(dataset, data_parallel_world_size, data_parallel_rank)
```

Comment:
Would you mind expanding the positional arguments into named ones for safety/clarity?

```python
DistributedSampler(dataset=dataset,
                   num_replicas=data_parallel_world_size,
                   rank=data_parallel_rank)
```
The formatter might have something to say about the line wrapping after that :-).

Reply:
Good call, just added the new commit!